health/health-checks/src/main/java/io/helidon/health/checks/DeadlockHealthCheck.java

```diff
@@ -80,17 +83,20 @@ public HealthCheckResponse call() {
                     .build();
         }
 
-        boolean noDeadLock = false;
+        HealthCheckResponseBuilder builder = HealthCheckResponse.builder()
+                .name(NAME);
+
         try {
             // Thanks to https://stackoverflow.com/questions/1102359/programmatic-deadlock-detection-in-java#1102410
-            noDeadLock = (threadBean.findDeadlockedThreads() == null);
+            long[] deadlockedThreads = threadBean.findDeadlockedThreads();
+            if (deadlockedThreads != null) {
+                builder.down();
```

Comment:
Returning `UP` is too optimistic, but returning `DOWN` is too pessimistic. We do not know that there is a deadlock (which is what `DOWN` should indicate); we cannot tell one way or the other.

Given there is no `ERROR` state available in the release, the code in this check should continue to throw an exception. 

The Helidon code which invokes this `call` method already catches exceptions and sets the "internalServerError` flag in the `HcResponse`. When the code in `HealthSupport#callHealthChecks` computes the overall status it should _ignore_ any `HcResponse` for which `internalServerError` has been set.

Reply:
Thank you for your explanation.

It's done. I'm not sure :) 

One small question for my curiosity:
If there is one internal error in response, then httpStatus=5xx and overallStatus = UP? Right?

Reply:
If there is _at least_ one internal error, then Helidon should return `500`. I guess the state in the response details might as well be `UP` (that's what happens in the updated 4.x code) but, ideally, K8s or whatever is monitoring the health endpoint will disregard the body of the response due to the HTTP status being 500.